openedx/core/djangoapps/edx_global_analytics/tasks.py

```diff
@@ -63,6 +64,42 @@ def enthusiast_level_statistics_bunch():
     return students_per_country
 
 
+def platform_coordinates(city_platform_located_in):
+    """
+    Method gets platform city latitude and longitude.
+
+    If `city_platform_located_in` (name of city) exists in OLGA setting (lms.env.json) as manual parameter
+    Google API helps to get city latitude and longitude. Else FreeGeoIP gathers latitude and longitude by IP address.
+
+    All correct city names are available from Wikipedia -
+    https://en.wikipedia.org/wiki/List_of_tz_database_time_zones
+
+    Module `pytz` also has list of cities with `pytz.all_timezones`.
+    """
+    def get_coordinates_by_ip():
+        """
+        Gather coordinates by server IP address with FreeGeoIP service.
+        """
+        ip_data = requests.get('http://freegeoip.net/json')
```

Comment:
Re: exceptions. I see you're handling them higher the call stack.
It's better to handle them here and not burden callee with such details.